rts/motoko-rts/src/gc/mark_compact/bitmap.rs

```diff
@@ -167,8 +167,8 @@ impl BitmapIter {
 
         // Outer loop iterates 64-bit words
         loop {
-            // Inner loop iterates bits in the current word
-            while self.current_word != 0 {
+            // Inner conditional examines the least significant bit(s) in the current word
+            if self.current_word != 0 {
```

Comment:
I wonder if you can (carefully) refactor this to:

```
           // Inner conditional examines the least significant bit(s) in the current word
            if self.current_word != 0 {
                if self.current_word & 0b1 != 0 {
                    let bit_idx = self.current_bit_idx;
                    self.current_word >>= 1;
                    self.current_bit_idx += 1;
                    return bit_idx;
                } else {
                    let shift_amt = self.current_word.trailing_zeros();
                    self.current_word >>= shift_amt;
                    self.current_word >>= 1;
                    let bit_idx = self.current_bit_idx + shift_amt;
                    self.current_bit_idx = bit_idx + 1;
                    return bit_idx;
                }
            }

```
to this, pulling out the common continuation.
```
           // Inner conditional examines the least significant bit(s) in the current word
            if self.current_word != 0 {
                let bit_idx = self.current_bit_idx;
                if self.current_word & 0b1 == 0 {
                   let shift_amt = self.current_word.trailing_zeros();
                   self.current_word >>= shift_amt;
                   self.current_word >>= 1;
                   bit_idx += shift_amt;
                }
                self.current_word >>= 1;
                self.current_bit_idx += 1;
                return bit_idx;		
            }
 
```

Reply:
The next evolution will be:
``` diff
diff --git a/rts/motoko-rts/src/gc/mark_compact/bitmap.rs b/rts/motoko-rts/src/gc/mark_compact/bitmap.rs
index 0856533ed..172e34e43 100644
--- a/rts/motoko-rts/src/gc/mark_compact/bitmap.rs
+++ b/rts/motoko-rts/src/gc/mark_compact/bitmap.rs
@@ -167,21 +167,14 @@ impl BitmapIter {
 
         // Outer loop iterates 64-bit words
         loop {
-            // Inner conditional examines the least significant bit(s) in the current word
+            // Examine the least significant bit(s) in the current word
             if self.current_word != 0 {
-                if self.current_word & 0b1 != 0 {
-                    let bit_idx = self.current_bit_idx;
-                    self.current_word >>= 1;
-                    self.current_bit_idx += 1;
-                    return bit_idx;
-                } else {
-                    let shift_amt = self.current_word.trailing_zeros();
-                    self.current_word >>= shift_amt;
-                    self.current_word >>= 1;
-                    let bit_idx = self.current_bit_idx + shift_amt;
-                    self.current_bit_idx = bit_idx + 1;
-                    return bit_idx;
-                }
+                let shift_amt = self.current_word.trailing_zeros();
+                self.current_word >>= shift_amt;
+                self.current_word >>= 1;
+                let bit_idx = self.current_bit_idx + shift_amt;
+                self.current_bit_idx = bit_idx + 1;
+                return bit_idx;
             }
 
             // Move on to next word (always 64-bit boundary)
```
which is even simpler than your solution. But let's merge this.